### PR TITLE
[DOC] Update docs for 24.04.0 release [skip ci]

### PR DESCRIPTION
**Pull request by SurajAralihalli**
This PR updates download docs for 24.04.0 release.

Signed-off-by: Suraj Aralihalli [suraj.ara16@gmail.com](mailto:suraj.ara16@gmail.com)

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -42,17 +42,17 @@ The plugin is tested on the following architectures:
...
-		Databricks 10.4 ML LTS (GPU, Scala 2.12, Spark 3.2.1)
 		Databricks 11.3 ML LTS (GPU, Scala 2.12, Spark 3.3.0)
 		Databricks 12.2 ML LTS (GPU, Scala 2.12, Spark 3.3.2)
 		Databricks 13.3 ML LTS (GPU, Scala 2.12, Spark 3.4.1)
 	
-	Supported Dataproc versions:
+	Supported Dataproc versions (Debian/Ubuntu):
 		GCP Dataproc 2.0
 		GCP Dataproc 2.1
+		GCP Dataproc 2.2
```
Comment (SurajAralihalli) [inline docs/download.md:55]: [Dataproc 2.2](https://cloud.google.com/dataproc/docs/concepts/versioning/dataproc-release-2.2) uses Spark 3.5.0 (Debian 12 and Ubuntu 22). I've updated our init scripts to support NVIDIA drivers installation on latest Debian, Ubuntu versions. I ran churn example but didn't run the integration tests. Please let me know if we can add it in the list of supported runtimes.

Comment (sameerz) [reply docs/download.md:55]: Since we are not doing integration tests on Debian, I would not say that it is supported.

File: docs/download.md [hunk 2]
```diff
@@ -84,26 +84,21 @@ CUDA 11.8 through CUDA 12.0.
...
-* Added support for parse_url functionality.
-* Introducing Lazy Quantifier support for regular expression functions.
-* Added support for the format_number function.
-* Enhanced batching support for row-based bounded window functions.
+* New functionality and performance improvements for this release include:
+* Enhanced decompression kernels for zstd and snappy.
+* Enhanced Parquet reading performance with modular kernels.
+* Added compatibility with Spark version 3.5.1.
+* Deprecated support for Databricks 10.4 ML LTS.
+* Added a new kernel for get_json_object, aiming to closely match Vanilla Spark output. This feature is disabled by default.
```
Comment (sameerz) [inline docs/download.md:101]: Not even sure if we should mention get_json_object yet, let's see how the open 24.04 get resolved or moved to 24.06.  

```suggestion
* Added a new kernel for get_json_object, aiming to match Spark output. This feature is disabled by default.
```

File: docs/download.md [hunk 3]
```diff
@@ -84,26 +84,21 @@ CUDA 11.8 through CUDA 12.0.
...
-* Introduced support for Databricks 13.3 ML LTS.
-* Added support for parse_url functionality.
-* Introducing Lazy Quantifier support for regular expression functions.
-* Added support for the format_number function.
-* Enhanced batching support for row-based bounded window functions.
+* New functionality and performance improvements for this release include:
+* Enhanced decompression kernels for zstd and snappy.
+* Enhanced Parquet reading performance with modular kernels.
+* Added compatibility with Spark version 3.5.1.
+* Deprecated support for Databricks 10.4 ML LTS.
```
Comment (sameerz) [inline docs/download.md:100]: Can we include a link to the new S3 perfio config? 
```suggestion
* Performance improvements for S3 reading 
* Performance improvements when doing a joins on unique keys
* Enhanced decompression kernels for zstd and snappy.
* Enhanced Parquet reading performance with modular kernels.
* Added compatibility with Spark version 3.5.1.
* Deprecated support for Databricks 10.4 ML LTS.
```

File: docs/download.md (hunk 1 above)
Comment (SurajAralihalli) [reply docs/download.md:55]: Noted, thanks!

**Review by NvTimLiu — COMMENTED** [review]
We have spark334 and spark342 shims, need we declare supporting them here?

File: docs/download.md [hunk 4]
```diff
@@ -42,17 +42,17 @@ The plugin is tested on the following architectures:
 		Apache Spark 3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.2.4
 		Apache Spark 3.3.0, 3.3.1, 3.3.2, 3.3.3
```
Comment (NvTimLiu) [inline docs/download.md:43]: We have spark334 and spark342 shims, need we declare supporting them here?

**Review by sameerz — APPROVED** [review]

**Review by NvTimLiu — APPROVED** [review]

**Comment by NvTimLiu** [comment]
build

**Pull request merged**